src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationFactory.cs

```diff
@@ -0,0 +1,313 @@
+        }
+
+        /// <summary>
+        /// Gets the assemblies containing the functional tests where the
+        /// <see cref="WebApplicationFactoryContentRootAttribute"/> define
+        /// content root to use for the given <typeparamref name="TEntryPoint"/>.
+        /// </summary>
+        /// <returns></returns>
+        protected virtual IEnumerable<Assembly> GetTestAssemblies()
+        {
+            try
+            {
+                // The default dependency context will be populated in .net core applications.
+                var context = DependencyContext.Default;
+                if (context != null)
+                {
+                    // Find the list of projects
+                    var projects = context.CompileLibraries.Where(l => l.Type == "project");
+
+                    // Find the list of projects runtime information and their assembly names.
+                    var runtimeProjectLibraries = context.RuntimeLibraries
+                        .Where(r => projects.Any(p => p.Name == r.Name))
+                        .ToDictionary(r => r, r => r.GetDefaultAssemblyNames(context).ToArray());
+
+                    var entryPointAssemblyName = typeof(TEntryPoint).Assembly.GetName().Name;
+
+                    // Find the project containing TEntryPoint
+                    var entryPointRuntimeLibrary = runtimeProjectLibraries
+                        .Single(rpl => rpl.Value.Any(a => string.Equals(a.Name, entryPointAssemblyName, StringComparison.Ordinal)));
+
+                    // Find the list of projects referencing TEntryPoint.
+                    var candidates = runtimeProjectLibraries
+                        .Where(rpl => rpl.Key.Dependencies
+                            .Any(d => string.Equals(d.Name, entryPointRuntimeLibrary.Key.Name, StringComparison.Ordinal)));
+
+                    return candidates.SelectMany(rl => rl.Value).Select(Assembly.Load);
+                }
+                else
+                {
+                    // The app domain friendly name will be populated in full framework.
+                    return new[] { Assembly.Load(AppDomain.CurrentDomain.FriendlyName) };
+                }
+            }
+            catch (Exception)
+            {
+            }
+
+            return Array.Empty<Assembly>();
+        }
+
+        private void EnsureDepsFile()
+        {
+            var depsFileName = $"{typeof(TEntryPoint).Assembly.GetName().Name}.deps.json";
+            var depsFile = new FileInfo(Path.Combine(AppContext.BaseDirectory, depsFileName));
+            if (!depsFile.Exists)
+            {
+                throw new InvalidOperationException(Resources.FormatMissingDepsFile(
+                    depsFile.FullName,
+                    Path.GetFileName(depsFile.FullName)));
+            }
+        }
+
+        /// <summary>
+        /// Creates a <see cref="IWebHostBuilder"/> used to setup <see cref="TestServer"/>.
+        /// <remarks>
```

Comment:
`<remarks>` is a *sibling* to `<summary>`, not a child. https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/remarks

Please check other places to see if this is a problem elsewhere.